### PR TITLE
Switch to cargo nextest as test runner

### LINKED ISSUE #2040: Investigate using nextest for test runs
**Issue summary**

There is an interesting tool - [cargo nextest](https://nexte.st/index.html). So far it looks promising; the test time is cut from 129 to 82s, and the report is more readable and seems nicely customizable. On top of this, it highlights slow tests.

image

Seems like a drop-in replacement but there may be some caveats. The goal here is to investigate this subject more, try to implement it and decide whether it's reasonable to run it instead of `cargo run`. 

**Other information and links**

### PR DESCRIPTION
**Pull request by creativcoder**
**Summary of changes**
Changes introduced in this pull request:
- Changes to Makefile to use cargo nextest as test runner

**Reference issue to close (if applicable)**

Closes #2040 

**Other information and links**

### CODE REVIEW

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
Could you also update the documentation?
https://github.com/chainsafe/forest/blob/creativcoder/faster-tests/README.md?plain=1#L139
https://github.com/chainsafe/forest/blob/creativcoder/faster-tests/README.md?plain=1#L148

**Comment by creativcoder** [comment]
Updated README.md

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
       - name: Cargo Check
         run: cargo check
       - name: Make Test-All
+        uses: taiki-e/install-action@nextest
```
Comment (hanabi1224) [inline .github/workflows/rust.yml:39]: Combination of uses and run does not seem to be valid

Comment (creativcoder) [reply .github/workflows/rust.yml:39]: I see, there's no error msg being reported nor docs to understand what's the issue is. So do I need to move all makefile test commands under the `run:` section instead?

Comment (creativcoder) [reply .github/workflows/rust.yml:39]: according to https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun i might just use run to install nextest and run make test

Comment (hanabi1224) [reply .github/workflows/rust.yml:39]: If u look at the CI jobs, `rust.yml` did not run

Comment (LesnyRumcajs) [reply .github/workflows/rust.yml:39]: @creativcoder Just put
```
- name: install nextest
   uses: taiki-e/install-action@nextest
```
above the cache step, leave this step as it was; it should work. To my understanding whether you can use `run` or not, depends on the plugin you are using. Here apparently it doesn't work.

Comment (creativcoder) [reply .github/workflows/rust.yml]: got it thanks! it's running now

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**